### PR TITLE
refactor: add functions to get parameters and add constants

### PR DESCRIPTION
**Pull request by aramase**
Signed-off-by: Anish Ramasekar <anish.ramasekar@gmail.com>

**Reason for Change**:

- Adds functions to get parameters from the attributes map
- Adds unit tests to validate the get parameters functions
- Makes the parameter names constant, so it can be reused in code and tests

**Requirements**

- [x] squashed commits
- [ ] included documentation
- [x] added unit tests and e2e tests (if applicable).

**Issue Fixed**:

**Does this change contain code from or inspired by another project?**

- [ ] Yes
- [x] No

**If "Yes," did you notify that project's maintainers and provide attribution?**

**Special Notes for Reviewers**:

### CODE REVIEW

File: pkg/provider/types/parameters.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+import (
+	"strconv"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+)
+
+// KeyVaultObject holds keyvault object related config
+type KeyVaultObject struct {
```
Comment (nilekhc) [inline pkg/provider/types/parameters.go:11]: should we rename constants.go -> `types.go` and put all structs and const in it? For eg https://github.com/kubernetes/client-go/blob/master/tools/clientcmd/api/types.go

I have seen this pattern in other k8s repos as well and personally follow the same.

File: pkg/provider/types/parameters_test.go [hunk 2]
```diff
@@ -0,0 +1,499 @@
+package types
+
+import (
+	"reflect"
+	"testing"
+)
+
+func TestGetKeyVaultName(t *testing.T) {
+	tests := []struct {
```
Comment (nilekhc) [inline pkg/provider/types/parameters_test.go:9]: This method does trimming. Should we add another test case with whitespaces and validate for Trim?

File: pkg/provider/types/parameters_test.go [hunk 3]
```diff
@@ -0,0 +1,499 @@
...
+			actual := GetKeyVaultName(test.parameters)
+			if actual != test.expected {
+				t.Errorf("GetKeyVaultName() = %v, expected %v", actual, test.expected)
+			}
+		})
+	}
+}
+
+func TestGetCloudName(t *testing.T) {
+	tests := []struct {
```
Comment (nilekhc) [inline pkg/provider/types/parameters_test.go:41]: same as above for Trim.

File: pkg/provider/types/parameters_test.go [hunk 4]
```diff
@@ -0,0 +1,499 @@
...
+	parameters := map[string]string{
+		"useVMManagedIdentity": "test",
+	}
+	if _, err := GetUseVMManagedIdentity(parameters); err == nil {
+		t.Errorf("GetUseVMManagedIdentity() error = nil, expected error")
+	}
+}
+
+func TestGetUserAssignedIdentityID(t *testing.T) {
+	tests := []struct {
```
Comment (nilekhc) [inline pkg/provider/types/parameters_test.go:203]: same as above. Could you also check for remaining tests?

File: pkg/provider/types/parameters_test.go (hunk 2 above)
Comment (aramase) [reply pkg/provider/types/parameters_test.go:9]: good call! added tests for trim

File: pkg/provider/types/parameters.go (hunk 1 above)
Comment (aramase) [reply pkg/provider/types/parameters.go]: I have no strong preference for this. I was using the kubelet package in k/k as ref which has constants in `constants.go` and we used that in other projects: https://github.com/kubernetes/kubernetes/tree/master/pkg/kubelet/types

I've move the structs to a single file and renamed it to `types.go` here.

**Review by nilekhc — APPROVED** [review]

**Pull request merged**